src/charm.py

```diff
@@ -37,6 +37,26 @@
 PEERS_RELATION_ENDPOINT_NAME = "peers"
 
 
+class TempoCoordinator(Coordinator):
+    """A Tempo coordinator class that inherits from the Coordinator class."""
+
+    @property
+    def _charm_tracing_receivers_urls(self) -> Dict[str, str]:
+        """Override with Tempo's enabled and requested receivers."""
+        # if Tempo is related to another Tempo instance, use the endpoints of that instance instead
+        if self.charm_tracing.is_ready():
+            return super()._charm_tracing_receivers_urls
+        return self._charm.requested_receivers_urls()
```

Comment:
I'd add a comment here to explain that we're returning 'this instance's endpoints' instead